src/Mod/PartDesign/App/FeatureRevolution.cpp

```diff
@@ -117,13 +129,23 @@ App::DocumentObjectExecReturn *Revolution::execute(void)
         if (sketchshape.IsNull())
             return new App::DocumentObjectExecReturn("Creating a face from sketch failed");
 
+        std::string method(Type.getValueAsString());

```

Comment:
You shouldn't repeat the same flaws from the pad/pocket feature. Instead of a std::string to handle the different methods use an enum class because this is faster and more secure because in case of a typo the compiler will complain.

Reply:
Sounds right. But I notice that the enumeration is also a property that can be changed outside of the task view. Do we sacrifice that or are there other options?

Reply:
@wwmayer do you have a suggestion of what to do about the enum being a property? It's straightforward to make an enum class instead of strings, but I need to know what to do about this being a property. Is there an example where `enum class` was used with `App::PropertyEnumeration`?

Reply:
> do you have a suggestion of what to do about the enum being a property? ...

I am not talking about the PropertyEnumeration using an enum. There you still have to use a list of strings. I am talking about the feature's execute() method where you use a string to handle the different modes.
```
std::string method = ... // assigned from property
if (method == "Method1") {
...
}
else if (method == "Method2") {
...
}
```

Now a string comparison is slow and error-prone because if you make a typo and write
```
if (method == "MEthod1") { // should be Method1
...
}
else if (method == "Method2") {
...
}
```
then the compiler won't be able to find the bug. You are only able to find that at runtime. Now my suggestion is to convert the string of the property to an enum and therefore you need a helper function:
```
enum class Method {
    Mehod1,
    Method2
};

Method methodFromString(const std::string& method)
{
    if (method == "Method1")
        return Method::Method1;
    if (method == "Method2")
        return Method::Method2;

    throw Base::ValueError("No such method");
}

void Feature::execute()
{
    Method method = methodFromString(enumProperty.getValue());
    if (method == Method::MEthod1) { // this typo will cause a build failure
...
    }
    else if (method == Method::Method2) {
...
    }
}
```

Of course, you still need a function where you compare strings but it's only one single function and not several functions and thus the likelihood of a typo is much lower.

Reply:
Thanks. I could probably reuse these enums in `TaskRevolutionParameters` as well.

Reply:
This should be done now. It's all still pretty messy, and groove is now completely broken.